app/controllers/home_controller.rb

```diff
@@ -1,30 +1,28 @@
 class HomeController < ApplicationController
+  before_action :authorize
+
   def dashboard
-    if user_signed_in?
       filter_users_status
       @room_messages = Room.global_chat_room.room_messages
       render 'dashboard'
-    else
-      render 'index'
-    end
-  end
-
-  def filter_users_status
-    @users = User.where.not(id: current_user.id).with_status(User.filter_status)
-    @users = @users.sample(maximum_length_user_list)
   end
 
   def chat
-    if user_signed_in?
-      @room_messages = Room.global_chat_room.room_messages
-      render 'home/chat_page'
-    else
-      render 'index'
-    end
+    @room_messages = Room.global_chat_room.room_messages
+    render 'home/chat_page'
   end
 
   private
 
+  def authorize
+    return render 'index' unless user_signed_in?
+  end
+
+  def filter_users_status
+    @users = User.where.not(id: current_user.id).with_status(User.filter_status)
+    @users = @users.sample(maximum_length_user_list)
+  end
+
   def maximum_length_user_list
     30
   end
```

Comment:
This will result in a merge conflict. But you should be able to handle it. maximum_length_user_list is a field now.

Reply:
This is resolved properly (by you, so I just resolve this :sweat_smile: )